lang/ir/src/ir/item_impl/constructor.rs

```diff
@@ -586,6 +553,12 @@ mod tests {
                 #[ink(event)]
                 fn my_constructor() -> Self {}
             },
+            // constructor + payable
+            syn::parse_quote! {
+                #[ink(constructor)]
+                #[ink(payable)]
```

Comment:
Should we allow `payable` on constructors now, since it is payable by default it could be misleading

Reply:
Payable constructors are actually not allowed and specifying this is a violation.